build/gen-deb.sh

```diff
@@ -0,0 +1,67 @@
+set -e
+
+if [ ! -f "opa_linux_amd64" ]; then
+  echo "ERROR: File 'opa_linux_amd64' not found. First build it 'make build-linux'."
+  exit 1
+fi
+
+if [ ! -d "man" ]; then
+  echo "ERROR: The man pages not found. First build them 'make man'."
+  exit 1
+fi
+
+if [ -z "$VERSION" ]; then
+  echo "ERROR: Need to have VERSION environment variable set."
+  exit 1
+fi
+
+rm -rf tmp_working_dir_opa
+mkdir tmp_working_dir_opa
+cd tmp_working_dir_opa
+
+mkdir -p ./usr/bin/
+cp ../opa_linux_amd64 ./usr/bin/opa
+chmod +x ./usr/bin/opa
+
+mkdir -p ./usr/share/man/man1/
+for MAN_PAGE in ../man/*.1; do
+  gzip --keep --best $MAN_PAGE
+done
+mv ../man/*.1.gz ./usr/share/man/man1/
+
+echo "Package: opa" > control
+echo "Version: $VERSION" >> control
+cat << 'EOF' >> control
+Architecture: amd64
+Maintainer: OPA Team <team@openpolicyagent.org>
+Depends: libc6 (>= 2.2.5)
+Section: admin
+Priority: optional
+Homepage: https://openpolicyagent.org
+Description: An open source, general-purpose policy engine.
+ The Open Policy Agent (OPA) is an open source, general-purpose
+ policy engine that enables unified, context-aware policy
+ enforcement across the entire stack.
+EOF
+
+md5sum ../opa_linux_amd64 > md5sums
+
+if command -v dpkg-deb >/dev/null 2>/dev/null; then
+  mkdir -p opa_$VERSION/DEBIAN
+  mv control opa_$VERSION/DEBIAN/control
+  mv md5sums opa_$VERSION/DEBIAN/md5sums
+  mv ./usr opa_$VERSION/usr
+
+  dpkg-deb -b opa_$VERSION/
+else # We will do this manually
+  echo 2.0 > debian-binary
+
+  tar cJf data.tar.xz ./usr/bin/opa ./usr/share/man/*
+  tar cJf control.tar.xz control md5sums
+
+  ar r opa_$VERSION.deb debian-binary control.tar.xz data.tar.xz
+fi
```

Comment:
I'd prefer we only had one path for this, aside from arguably be the "right way" is there any benefit to using `dpkg-deb` over the more manual archive creation?

Would it make sense to just assume we have `dpgk-deb` and have the make targets rely on running in a container which contains the tooling required?

Reply:
Assuming `dpkg-deb` is probably the best route. If it tries to get built on a different container/host system that does not have `dpkg-deb` that would be a problem and hence the manual way. However, I think it is ok to reduce complexity in exchange for a simple comment that says we assume `dpkg-deb` to be there.

Reply:
Removed the branch and added a check to ensure we have `dpkg-deb` before script continues.